internal/mode/static/state/dataplane/configuration.go

```diff
@@ -852,6 +852,26 @@ func buildBaseHTTPConfig(g *graph.Graph) BaseHTTPConfig {
 		}
 	}
 
+	if g.NginxProxy.Source.Spec.RewriteClientIP != nil {
+		if g.NginxProxy.Source.Spec.RewriteClientIP.Mode != nil {
+			switch *g.NginxProxy.Source.Spec.RewriteClientIP.Mode {
+			case ngfAPI.RewriteClientIPModeProxyProtocol:
+				baseConfig.RewriteClientIPSettings.Mode = RewriteIPModeProxyProtocol
+			case ngfAPI.RewriteClientIPModeXForwardedFor:
+				baseConfig.RewriteClientIPSettings.Mode = RewriteIPModeXForwardedFor
+			}
+		}
+
+		if len(g.NginxProxy.Source.Spec.RewriteClientIP.TrustedAddresses) > 0 {
+			trustedAddresses := convertTrustedAddresses(g)
+			baseConfig.RewriteClientIPSettings.TrustedAddresses = trustedAddresses
```

Comment:
nit: better to combine and save a variable:

```suggestion
			baseConfig.RewriteClientIPSettings.TrustedAddresses = convertTrustedAddresses(g)
```

Reply:
unresolving cause this suggestion wasn't applied